src/WebJobs.Script.WebHost/Security/SecretGenerator.cs

```diff
@@ -45,5 +60,22 @@ internal static string GenerateIdentifiableSecret(ulong seed)
             // equal signs do not require encoding.
             return IdentifiableSecrets.GenerateUrlSafeBase64Key(seed, 40, AzureFunctionsSignature, elidePadding: false);
         }
+
+        internal static bool ValidateSecret(string key, ulong seed)
+        {
+            return IdentifiableSecrets.ValidateBase64Key(key, seed, SecretGenerator.AzureFunctionsSignature, encodeForUrl: true);
+        }
+
+        internal static bool TryValidateSecret(string key, ulong seed)
```

Comment:
Could we update this API to take a secret type instead of a seed? That would simplify usage as it wouldn't require the resolution from secret type to a seed first (or having knowledge of what seed to use).

Once that is in place, you might be able to completely remove `GetKeyType`

Reply:
I looked at this and making that change will result in a bunch of cascading changes that I don't want to make right now. Can consider as a future refactoring.